### DISCUSSION TITLE
`npx` command not found when calling `exec` in test

### DISCUSSION THREAD

**Discussion by jonasmerlin** [avajs/ava#2948] [Q&A] [answered]
Hey, so I'm trying to run my integration tests in parallel against a real DB. For this, I'm using a different schema for each of my test files. Each of these needs to be migrated to reflect my schema. Since I'm using prisma, where migrate is only a CLI command, I try to run it via `execSync()` from node's own `child_process` module. So, all in all, this is roughly what I'm doing:

```javascript
test.before(async (t) => {
  const dbUrl = `postgresql://admin:test@localhost:5432/db?schema=pluginPrisma&connection_limit=1`;

  execSync("npx prisma migrate dev", {
    env: {
      DATABASE_URL: dbUrl,
    },
  });

  // ... rest of before
}
```

But I'm getting the following error when running my tests:

```
  before hook

  Rejected promise returned by test. Reason:

  Error {
    output: [
      null,
      Buffer @Uint8Array [],
      Buffer @Uint8Array [
        2f62696e 2f73683a 206e7078 3a20636f 6d6d616e 64206e6f 7420666f 756e640a
      ],
    ],
    pid: 7087,
    signal: null,
    status: 127,
    stderr: Buffer @Uint8Array [
      2f62696e 2f73683a 206e7078 3a20636f 6d6d616e 64206e6f 7420666f 756e640a
    ],
    stdout: Buffer @Uint8Array [],
    message: `Command failed: npx prisma migrate dev␊
    /bin/sh: npx: command not found␊
    `,
  }

  › /bin/sh: npx: command not found
  › __tests__/plugins/prisma.test.ts:12:11

  ─
```

Same goes for running `npm run <something>` . But *running stuff like `ls` works*!

I'm using ava 4.0.1 and guess this has something to do with the thread isolation. Any ideas on how to work around this?

Thanks!

**Comment by novemberborn** [comment, marked as answer]
My guess is that the way you provide the env object your other environment variables may not be passed through, meaning that PATH is empty and the child process can't find npx. ls is built-in so works. I don't expect this to be related to the worker threads.

**Reply by jonasmerlin** [reply]
Thanks for your quick answer. This is exactly what was happening, of course. Thank you!
